addons/io/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java

```diff
@@ -330,7 +343,14 @@ private void handleRequestEvent(JSONObject data) {
             logger.debug("Request method is {}", requestMethod);
             Request request = jettyClient.newRequest(requestUri);
             setRequestHeaders(request, requestHeadersJson);
-            request.header("X-Forwarded-Proto", "https");
+            if (protocol != null) {
+                request.header("X-Forwarded-Proto", protocol);
+            } else if (data.has("protocol")) {
+                request.header("X-Forwarded-Proto", data.getString("protocol"));
```

Comment:
How did you decide the order, why is the protocol from the base url the most important and then this one? If you would give the `data.getString('protocol')` precendence your code could be simpler, by just setting the protocol to https in the catch block.